Controllers/AdminController.js

```diff
@@ -0,0 +1,177 @@
+const AsyncHandler = require("express-async-handler");
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const Admin = require("../Models/AdminModel");
+const Book = require("../Models/BookModel");
+const Student = require("../Models/StudentModel");
+const DefinedError = require("../Middleware/DefinedError");
+const { errHandle } = require("../Middleware/errHandle");
+
+const RegisterUser = async (req, res) => {
+  try {
+    const { username, password, name } = req.body;
+
+    if (!username || !password || !name) {
+      throw new DefinedError(
+        400,
+        "error",
+        "Please Add All Fields",
+        "Admin Not Registered"
+      );
+    }
+
+    const UserExists = await Admin.findOne({ username });
```

Comment:
```suggestion
    const user = await Admin.findOne({ username });
```
var name starts with small letter
and the name UserExists should have a boolean value but we have an object here so always its user